File: app/src/main/java/com/innso/mobile/ui/viewModels/ProfileViewModel.java
package com.innso.mobile.ui.viewModels;


import android.view.View;

import com.google.firebase.auth.FirebaseAuth;
import com.innso.mobile.ui.activities.CreateUserActivity;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.subjects.PublishSubject;

public class ProfileViewModel extends BaseViewModel {


    private PublishSubject<Boolean> closeSession = PublishSubject.create();

    @Inject
    FirebaseAuth firebaseAuth;

    public ProfileViewModel() {
        getComponent().inject(this);
    }

    public void addUser(View view){
        startActivityEvent.onNext(CreateUserActivity.class);
    }

    public void addCustomer(View view){
        startActivityEvent.onNext(CreateUserActivity.class);
    }

    public void closeSession(View view){
        firebaseAuth.signOut();
        closeSession.onNext(true);
    }

    public Observable<Boolean> closeSession(){
        return closeSession.observeOn(AndroidSchedulers.mainThread());
    }

}

File: app/src/main/java/com/innso/mobile/viewModels/company/ProfileViewModel.kt
package com.innso.mobile.viewModels.company


import android.view.View
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.google.firebase.auth.FirebaseAuth
import com.innso.mobile.viewModels.AndroidViewModel
import javax.inject.Inject

class ProfileViewModel : AndroidViewModel() {

    val userInformation: MutableLiveData<String> = MutableLiveData()

    private val closeSession: MutableLiveData<Unit> = MutableLiveData()

    @Inject
    lateinit var firebaseAuth: FirebaseAuth

    init {
        getComponent().inject(this)
        initUserInfo()
    }

    fun initUserInfo() {
        firebaseAuth.currentUser?.let {
            userInformation.value = it.displayName + "\n" + it.email
        }
    }

    fun closeSession(view: View) {
        firebaseAuth.signOut()
        closeSession.postValue(Unit)
    }

    fun closeSession(): LiveData<Unit> = closeSession

}
